http/src/main/java/io/micronaut/http/HttpRequest.java

```
@@ -198,6 +198,13 @@ default boolean isSecure() {
         return this.getAttribute(HttpAttributes.X509_CERTIFICATE, Certificate.class);
     }
 
+    default boolean isServerPushSupported() {
+        return false;
+    }
+
+    default void serverPush(HttpRequest<?> request) {
```

Should probably throw `UnsupportedOperationException` if it is unsupported by the impl

---

My thought was that these apis are "best-effort" anyway and should fail silently if unsupported. But maybe not, because you want to avoid the computation of the parameters if the client doesn't support server push anyway.

---

there is already a method to check if push is supported. If the developer calls this first they can still implement best effort without being confused why it doesn't work